src/exportStrategies.js

```diff
@@ -116,8 +116,9 @@ export const CSVStream = async ({
   logger = console.info,
 }) => {
   let records = 0
-  let columnKeys = []
   let totalRecords = await strategy.getTotalRecords()
+  let includeColumns = _.getOr([], 'include', strategy)
```

Comment:
The only change that is needed in this entire file is:
```
let columnKeys = _.getOr([], 'include', strategy)
```
instead of
```
let columnKeys = []
```
everything else should stay the same.

Reply:
No that is not true at all. When you "only" do this in this file what happens is the following:

1. `columnKeys` is no longer empty so we will skip the attempts to get the column names from the file first line so far great!
2. `let rows = convertData(formattedData, columnKeys)` will fail since the formattedData already has the columns in `start case` format and our "keys" are still in their "key" format passed through.

![image](https://user-images.githubusercontent.com/910303/57538976-554ed480-72fe-11e9-9c62-95fb2e2bb872.png)

3. Going further you can already see the issue:
`rows = [convertColumns(columnKeys), ...rows]` would now have this:

![image](https://user-images.githubusercontent.com/910303/57539271-005f8e00-72ff-11e9-9de7-88e5d08185d7.png)

Which is due to the fact that the columns which were space cased did not match with the ones "just passed". Therefore that is not at all the only change needed in this file. I did try this first btw.

Reply:
I'll reach out via Slack.

Reply:
I think I understand. Sounds like you just need to do:
```js
let columnKeys = convertColumns(_.getOr([], 'include', strategy))
```
and change
```js
        rows = [convertColumns(columnKeys), ...rows]
```
to
```js
        rows = [columnKeys, ...rows]
```
since the `convertColumns` call is unnecessary there.

Reply:
Well this is what I am doing essentially the only difference being I separate `keys` and `headers` which I think is more readable. Now you know you are having `includedKeys` coming in and then you have `columnHeaders`. Also I separate and comment on the case where we `extractKeysFromFirstRow` via:

```
      // If no includedKeys ware passed get them from the first row
      // this is not accurate and only works in the case where the first row has all the data for all columns
      if (_.isEmpty(includedKeys)) {
        // Extract column names from first object
        columnHeaders = extractKeysFromFirstRow(formattedData)
      } else {
        // Start Case the passed includedKeys
        columnHeaders = convertColumns(includedKeys)
      }
```

To me that is much more readable than this one liner:
`let columnKeys = convertColumns(_.getOr([], 'include', strategy))`

Also in the case of `convertData` it really should accept headers since it deals with the row data. 

I agree on the `convertColumns` case that it should accept `keys` and I changed that. Let me know if you disagree on the readability argument. I think seeing this the first time would be easier to comprehend than the one liner (which I also had before).

Reply:
I don't prefer `columnHeaders` here since we are using `columnKeys` everywhere and that seems to capture the idea more precisely.

Reply:
Ok but look at the latest changes I did. 

`convertColumns` works with column keys but `convertData` with headers.
One extracts the headers from the keys and he other deals with the rows and works with headers. I think this is easier to read. It seems to be very much a personal preference at this point really. Lets see if others can chime in.

Reply:
Right - hence my comment that we shouldn't be renaming the keys to be the headers on the actual data. That's a lot of extra work to be doing - changing that will both make it more efficient and make the naming question moot because it actually will be keys and not headers.